libbpf-rs/src/map.rs

```diff
@@ -163,22 +163,37 @@ pub struct Map {
 }
 
 impl Map {
-    pub(crate) fn new(
-        fd: i32,
-        name: String,
-        ty: libbpf_sys::bpf_map_type,
-        key_size: u32,
-        value_size: u32,
-        ptr: NonNull<libbpf_sys::bpf_map>,
-    ) -> Self {
-        Map {
+    /// Create a [`Map`] from a [`libbpf_sys::bpf_map`].
+    ///
+    /// # Safety
+    ///
+    /// The pointer must point to a loaded map.
+    pub(crate) unsafe fn new(ptr: NonNull<libbpf_sys::bpf_map>) -> Result<Self> {
+        // Get the map name
+        // SAFETY:
+        // bpf_map__name can return null but only if it's passed a null.
```

Comment:
It appears you are describing properties of the output, less so of the input. But that has not relevance to safety itself, because it is up to the callee. I suppose that should ultimately go to a safety comment  on `c_ptr_to_string`, which really should be `unsafe`.

Reply:
Yeah, I agree the safety comment here pertains more to the `c_ptr_to_string` call. I would mark that function as unsafe but it's used a lot so it would blow up the size of the diff. 

This comment was copied from where this code was moved from, so I just left it as it. Although it isn't perfect it does help understand why this bit of code is okay